motionplan/planManager.go

```diff
@@ -289,26 +293,26 @@ func (pm *planManager) planSingleAtomicWaypoint(
 		case nextSeed := <-endpointPreview:
 			return nextSeed.Q(), &resultPromise{future: solutionChan}, nil
 		case planReturn := <-solutionChan:
-			if planReturn.planerr != nil {
-				return nil, nil, planReturn.planerr
+			if planReturn.err != nil {
+				return nil, nil, planReturn.err
 			}
-			steps := nodesToInputs(planReturn.steps)
-			return steps[len(steps)-1], &resultPromise{steps: steps}, nil
+			seed := planReturn.steps[len(planReturn.steps)-1].Q()
```

Comment:
Is this fixing a bug?

Reply:
I dont understand, this looks like a 1:1 refactor to me.  Maybe I'm missing something

Reply:
the `Q()` part threw me off fi didn't understand what that was doing